dataprofiler/profilers/unstructured_text_profile.py

```diff
@@ -251,13 +249,17 @@ def _update_words(self, data, prev_dependent_properties=None,
         :type subset_properties: dict
         :return: None
         """
-        for row in data:
-            for word in re.findall(r'\w+', row):
-                word_lower = word.lower()
-                if word_lower not in self._stop_words:
-                    if not self._is_case_sensitive:
-                        word = word_lower
-                    self.word_count[word] += 1
+        if not self._is_case_sensitive:
+            linewords = ([w.strip(string.punctuation) for w in line.lower().split()]
+                         for line in data)
+        else:
+            linewords = ([w.strip(string.punctuation) for w in line.split()]
+                         for line in data)
+        word_count = Counter(itertools.chain.from_iterable(linewords))
+
+        for w, c in word_count.items():
+            if w.lower() not in self._stop_words:
+                self.word_count.update({w: c})
```

Comment:
Couldn't you filter stop words within the list comprehension? i.e. `for w in line.split() if w not in stop_words for line in data` (idk the order there may be wrong)